tokio-util/src/io/sink_writer.rs

```diff
@@ -0,0 +1,183 @@
+use bytes::Bytes;
+use futures_sink::Sink;
+
+use pin_project_lite::pin_project;
+use std::io;
+use std::pin::Pin;
+use std::task::{Context, Poll};
+use tokio::io::AsyncWrite;
+
+pin_project! {
+    /// Convert a [`Sink`] of byte chunks into an [`AsyncWrite`].
+    /// Bytes are sent into the sink and the sink is flushed once all bytes are sent. If an error occurs during the sending progress,
+    /// the number of sent but unflushed bytes are saved in case the flushing operation stays unsuccessful.
+    /// For the inverse operation of defining an [`AsyncWrite`] from a [`Sink`] you need to define a [`codec`].
+    ///
+    /// # Example
+    ///
+    /// ```
+    /// use futures_util::SinkExt;
+    /// use std::io::{Error, ErrorKind};
+    /// use tokio::io::AsyncWriteExt;
+    /// use tokio_util::io::{SinkWriter, CopyToBytes};
+    /// use tokio_util::sync::PollSender;
+    ///
+    /// # #[tokio::main]
+    /// # async fn main() -> Result<(), Error> {
+    ///  // Construct a channel pair to send data across and wrap a pollable sink.
+    ///  // Note that the sink must mimic a writable object, e.g. have `std::io::Error`
+    ///  // as its error type.
+    /// let (tx, mut rx) = tokio::sync::mpsc::channel::<Vec<u8>>(1);
+    ///  let mut writer = SinkWriter::new(CopyToBytes::new(
+    ///    PollSender::new(tx).sink_map_err(|_| Error::from(ErrorKind::BrokenPipe)),
+    /// ));
```

Comment:
I doubt that this example compiles.

Reply:
Would be too easy if I actually checked all code that I check in. My bad.